docs/src/man/basics.md

```diff
@@ -1565,40 +1565,1265 @@ julia> german[Not(5), r"S"]
+
+!!! Note
+      The Julia parser sometimes prevents `:` from being used by itself.
+      `ERROR: syntax: whitespace not allowed after ":" used for quoting`
+      means your `:` must be wrapped in either `(:)` or `Cols(:)`
+      to be properly interpreted.
+
+```julia
+julia> df = DataFrame(
+           id = [1, 2, 3],
+           first_name = ["José", "Emma", "Nathan"],
+           last_name = ["Garcia", "Marino", "Boyer"],
+           age = [61, 24, 33]
+       )
+3×4 DataFrame
+ Row │ id     first_name  last_name  age
+     │ Int64  String      String     Int64
+─────┼─────────────────────────────────────
+   1 │     1  José        Garcia        61
+   2 │     2  Emma        Marino        24
+   3 │     3  Nathan      Boyer         33
+
+julia> select(df, [:last_name, :first_name])
+3×2 DataFrame
+ Row │ last_name  first_name
+     │ String     String
+─────┼───────────────────────
+   1 │ Garcia     José
+   2 │ Marino     Emma
+   3 │ Boyer      Nathan
+
+julia> select(df, r"name")
+3×2 DataFrame
+ Row │ first_name  last_name
+     │ String      String
+─────┼───────────────────────
+   1 │ José        Garcia
+   2 │ Emma        Marino
+   3 │ Nathan      Boyer
+
+julia> select(df, Not(:id))
+3×3 DataFrame
+ Row │ first_name  last_name  age
+     │ String      String     Int64
+─────┼──────────────────────────────
+   1 │ José        Garcia        61
+   2 │ Emma        Marino        24
+   3 │ Nathan      Boyer         33
+
+julia> select(df, Between(2,4))
+3×3 DataFrame
+ Row │ first_name  last_name  age
+     │ String      String     Int64
+─────┼──────────────────────────────
+   1 │ José        Garcia        61
+   2 │ Emma        Marino        24
+   3 │ Nathan      Boyer         33
+
+julia> df2 = DataFrame(
+           name = ["Scott", "Jill", "Erica", "Jimmy"],
+           minor = [false, true, false, true],
+           male = [true, false, false, true],
+       )
+4×3 DataFrame
+ Row │ name    minor  male
+     │ String  Bool   Bool
+─────┼──────────────────────
+   1 │ Scott   false   true
+   2 │ Jill     true  false
+   3 │ Erica   false  false
+   4 │ Jimmy    true   true
+
+julia> subset(df2, [:minor, :male])
+1×3 DataFrame
+ Row │ name    minor  male
+     │ String  Bool   Bool
+─────┼─────────────────────
+   1 │ Jimmy    true  true
+```
+
+`AsTable(source_column_selector)` is a special `source_column_selector`
+that can be used to select multiple columns into a single `NamedTuple`.
+This is not useful on its own, so the function of this selector
+will be explained in the next section.
+
+
+#### `operation_function`
+Inside an `operation` pair, `operation_function` is a function
+which operates on data frame columns passed as vectors.
+When multiple columns are selected by `source_column_selector`,
+the `operation_function` will receive the columns as multiple positional arguments
+in the order they were selected, e.g. `f(column1, column2, column3)`.
```

Comment:
the `f` and `column1` etc. are undefined here so it is not clear what they refer to.

Reply:
I am just trying to give an idea of what it means for column vectors to be mapped to positional arguments without a full example (because I want to present the simplier ideas first). I added a more concrete example (function `h`) to the end of this section, but I would like to illustrate the jargon here somehow. To me, the names `f` and `columnx` are clear enough, but maybe there is something better I can do.

Reply:
OK. But that is why I put the comment at the beginning after seeing several examples of this style to make this page separate and clearly comment on its purpose. The thing is that this material is aimed to be more "intuitive" than the rest of the documentation, but it also means that it has a less formal style. I think it should be clearly signaled to the readers of the manual.

Reply:
I am okay if you have suggestions to make the content better match style and tone. The important aspect to me is laying out the concepts at a pace that a beginner can understand. In this specific example, I am not sure if the reader knows what "separate positional arguments" means yet, so I tried to simply illustrate. I am open to other ideas though.

Reply:
That is my point - you know what a beginner would understand 😄. I and @nalimilan can just make sure that what you write is technically correct (but I am OK with retaining your pedagogic style).

The issue is that I would assume that someone begins with DataFrames.jl but knows Julia. But the material you prepare is probably for people who do not know DataFrames.jl and are not experts on Julia.